### PR TITLE
Update SQL to use parameterised queries

### PR DESCRIPTION
**Pull request by craigfrancis**
It's really bad practice to concatenate user supplied variables into an SQL string. While the original did cast the variable to an int, it's very easy to make a mistake.

### CODE REVIEW

**Comment by cmb69** [comment]
Thanks for the PR!

See also the related [discussion on the mailing list](https://marc.info/?l=phpdoc&m=160658331129934&w=2).

File: reference/mysqli/examples.xml [hunk 1]
```diff
@@ -89,7 +93,10 @@ echo "Sometimes I see " . $actor['first_name'] . " " . $actor['last_name'] . " o
 // Now, let's fetch five random actors and output their names to a list.
 // We'll add less error handling here as you can do that on your own now
 $sql = "SELECT actor_id, first_name, last_name FROM actor ORDER BY rand() LIMIT 5";
-if (!$result = $mysqli->query($sql)) {
+if ($stmt = $mysqli->prepare($sql)) {
+    $stmt->execute();
+    $result = $stmt->get_result();
```
Comment (nikic) [inline reference/mysqli/examples.xml:98]: I don't think it makes sense to use prepare for a query without placeholders.

Comment (craigfrancis) [reply reference/mysqli/examples.xml:98]: I go on the basis that the query I write today will be edited by someone else later, and if they aren't familiar with prepared statements, by using `prepare()` for everything, there is a much better chance of a good outcome (that said, I wish prepare/bind_param/execute was easier to write).

**Comment by cmb69** [comment]
Frankly, I don't see the point in making this example even more complex than it is now. Of course, building queries with user supplied input using string interpolation/concatenation is bad practice, but I see this example as some kind of started (but never completed) tutorial for using MySQLi. Thus I still think that we should either drop this example section altogether, or that we should expand it to a tutorial. A first step might be to keep the example as is (except some of the comments), and then follow up pointing out serious issues with the example, and showing better solutions (like using prepared statements or [proper output escaping](https://github.com/php/doc-en/pull/279).

**Comment by craigfrancis** [comment]
Unfortunately writing safe mysqli code is complex, and while I want that API to change, it won’t happen any time soon.

Today I see a page on the official PHP website, that new developers are copying, that teaches them really bad coding practices.

If you want to delete that page, fine, but do it now.

If you want to do a re-write, go for it, but if you’re going down this route, at least apply this small patch now.

There are comments in the code already, and they *will* be ignored (I’ve been in the unfortunate position of telling far too many programmers that the code they have written/copied is fundamentally flawed, and I’m really disappointed to find this unsafe example code here).

File: reference/mysqli/examples.xml (hunk 1 above)
Comment (kamil-tekiela) [reply reference/mysqli/examples.xml:98]: Ideally, everyone should use prepared statements 100% of the time. The only downside is that you have to add 2 more lines over `mysqli_query`. I agree that in a tutorial an example should be simple, but mysqli is not simple. If you want something simple then go for PDO. IMO that example fails to teach the proper mysqli usage. I already said that I am either way for improving or removing it altogether, but I would really recommend to remove it and the whole section from the manual.

**Comment by cmb69** [comment]
> If you want to delete that page, fine, but do it now.

I think that is the best solution. Can I please have a straw poll (👍 or 👎)?

**Comment by craigfrancis** [comment]
Thanks @cmb69, I think it looks pretty definitive (6 to delete).

I've also started a thread on PHP internals, and an RFC to hopefully address the difficulty in using parameterised queries in mysqli, https://news-web.php.net/php.internals/112618

**Comment by cmb69** [comment]
Thanks for the reminder. I have [removed the example](http://svn.php.net/viewvc?view=revision&revision=352184).

**Comment by kamil-tekiela** [comment]
Bug report updated. [#77531](https://bugs.php.net/bug.php?id=77531)

**Pull request closed without merging**